src/background.ts

```diff
@@ -87,16 +87,25 @@ async function sendTask(tabId: number, messageId: string, data: string): Promise
     return;
   }
 
-  try {
-    const hosts = await getHosts();
-    for (const host of hosts) {
-      try {
-        await host.send(data);
-      } catch (err) {
-        //
-      }
+  // Build requests to all hosts while firing them in parallel.
+  const hosts = await getHosts();
+  const requests = [];
+  for (const host of hosts) {
+    try {
+      requests.push(host.send(data));
+    } catch (err) {
+      //
     }
+  }
 
+  // Wait for all requests to finish.
+  try {
+    await Promise.allSettled(requests);
+  } catch (err) {
+    //
```

Comment:
This looks not right. Given that we are expecting at least 1 request to succeed, the error handling approach can be this:
1. Iterate through the output of `await Promise.allSettled(requests)`
2. exit gracefully when 1 success request is found
3. if no request succeeded, maybe show a popup "No clients found" or something like that?
Also this should be inside try block. Also handle error inside catch block.

Reply:
This matches the old error handling related to requests, so it seems fine to me. See https://github.com/jmerle/competitive-companion/pull/467#issuecomment-2161777600 for more information.